bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/discovery/BroadlinkThermostatDiscoveryService.java

```diff
@@ -0,0 +1,202 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.broadlinkthermostat.internal.BroadlinkThermostatBindingConstants;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.net.NetworkAddressService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.mob41.blapi.BLDevice;
+
+/**
+ * The {@link BroadlinkThermostatDiscoveryService} is responsible for discovering Broadlinkthermostat devices through
+ * Broadcast.
+ *
+ * @author Florian Mueller - Initial contribution
+ */
+@Component(service = DiscoveryService.class, configurationPid = "discovery.broadlinkthermostat")
+@NonNullByDefault
+public class BroadlinkThermostatDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(BroadlinkThermostatDiscoveryService.class);
+
+    private final NetworkAddressService networkAddressService;
+
+    private static final Set<ThingTypeUID> DISCOVERABLE_THING_TYPES_UIDS = Set.of(FLOUREON_THERMOSTAT_THING_TYPE,
+            UNKNOWN_BROADLINKTHERMOSTAT_THING_TYPE);
+    private static final int DISCOVERY_TIMEOUT = 30;
+    private final Runnable scanner;
+    private @Nullable ScheduledFuture<?> backgroundDiscoveryFuture;
+
+    @Activate
+    public BroadlinkThermostatDiscoveryService(@Reference NetworkAddressService networkAddressService) {
+        super(DISCOVERABLE_THING_TYPES_UIDS, DISCOVERY_TIMEOUT);
+        this.networkAddressService = networkAddressService;
+        scanner = createScanner();
+    }
+
+    private Runnable createScanner() {
+        return () -> {
+            long timestampOfLastScan = getTimestampOfLastScan();
+            BLDevice[] blDevices = new BLDevice[0];
+            try {
+                @Nullable
+                InetAddress sourceAddress = getIpAddress();
+                if (sourceAddress != null) {
+                    logger.debug("Using source address {} for sending out broadcast request.", sourceAddress);
+                    blDevices = BLDevice.discoverDevices(sourceAddress, 0, DISCOVERY_TIMEOUT * 1000);
+                } else {
+                    blDevices = BLDevice.discoverDevices(DISCOVERY_TIMEOUT * 1000);
+                }
+
```

Comment:
This unnecessary empty line is present in my checkstyle warnings.

Reply:
Ok, removed it.